core/src/main/java/io/grpc/ExponentialBackoffPolicy.java

```diff
@@ -0,0 +1,40 @@
+package io.grpc;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.Random;
+
+/**
+ * Retry Policy for Transport reconnection.
+ */
+class ExponentialBackoffPolicy {
+  /**
+   * Parameters from https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
+   */
+  static final double INITIAL_BACKOFF_MILLIS = 1 * 1000;
+  static final double MULTIPLIER = 1.6;
+  static final double MAX_BACKOFF_MILLIS = 2 * 60 * 1000;
+  static final double MIN_CONNECT_TIMEOUT_MILLIS = 20 * 1000;
```

Comment:
Leaving MIN_CONNECT_TIMEOUT param out entirely sounds good. At some time it may be good to limit the connection attempt time like the spec assumes, but that does seem like future work.